lib/active_model/serializer/collection_serializer.rb

```diff
@@ -32,8 +32,7 @@ def json_key
 
       def paginated?
         object.respond_to?(:current_page) &&
-          object.respond_to?(:total_pages) &&
-          object.respond_to?(:size)
```

Comment:
Here I assume that every collection should respond to `#size`. Correct me if I'm wrong.